src/memory/RomManbow2.cc

```diff
@@ -31,45 +31,29 @@ namespace openmsx {
 	}
 }
 
-using Info = AmdFlash::SectorInfo;
-
-// 512kB, only last 64kB writable
-static constexpr auto config1 = [] {
-	std::array<Info, 512 / 64> result = {};
-	ranges::fill(result, Info{64 * 1024, true}); // read-only
-	result[7].writeProtected = false;
-	return result;
-}();
-// 512kB, only 128kB writable
-static constexpr auto config2 = [] {
-	std::array<Info, 512 / 64> result = {};
-	ranges::fill(result, Info{64 * 1024, true}); // read-only
-	result[4].writeProtected = false;
-	result[5].writeProtected = false;
-	return result;
-}();
-// fully writeable, 512kB
-static constexpr auto config3 = [] {
-	std::array<Info, 512 / 64> result = {};
-	ranges::fill(result, Info{64 * 1024, false});
-	return result;
-}();
-// fully writeable, 2MB
-static constexpr auto config4 = [] {
-	std::array<Info, 2048 / 64> result = {};
-	ranges::fill(result, Info{64 * 1024, false});
-	return result;
-}();
-[[nodiscard]] static constexpr std::span<const Info> getSectorInfo(RomType type)
+[[nodiscard]] static const std::span<const AmdFlash::SectorInfo> getSectorInfo(RomType type)
 {
 	switch (type) {
 	using enum RomType;
 	case MANBOW2:
-	case MANBOW2_2:             return config1;
-	case HAMARAJANIGHT:         return config2;
-	case MEGAFLASHROMSCC:       return config3;
-	case RBSC_FLASH_KONAMI_SCC: return config4;
-	default: UNREACHABLE;
+	case MANBOW2_2:
+		{
+			static const std::array<const AmdFlash::SectorInfo, 8> sectorInfo =
+				{{{true}, {true}, {true}, {true}, {true}, {true}, {true}, {false}}};
+			return sectorInfo;  // only last 64kB writable
+		}
+	case HAMARAJANIGHT:
+		{
+			static const std::array<const AmdFlash::SectorInfo, 8> sectorInfo =
+				{{{true}, {true}, {true}, {true}, {false}, {false}, {true}, {true}}};
```

Comment:
Now that I see it written like this ... is it worth keeping the `SectorInfo` struct? It only contains one `bool`. Maybe just have an array/span of `bool`. And if we rename `sectorInfo` to something like `sectorsWriteProtected`, that's then equally clear?